substrate/frame/tips/src/lib.rs

```diff
@@ -465,6 +469,21 @@ pub mod pallet {
 			Ok(())
 		}
 	}
+
+	#[pallet::hooks]
+	impl<T: Config<I>, I: 'static> Hooks<BlockNumberFor<T>> for Pallet<T, I> {
+		fn integrity_test() {
+			assert!(
+				!T::TipReportDepositBase::get().is_zero(),
```

Comment:
Please add a comment to `TipReportDepositBase` that mentions this restriction.

Reply:
> Btw i am not sure if we should even restrict it.

Depends on the likelihood a user would not set `DataDepositPerByte` to Zero, which would result in a zero finders fee on `report_awesome` when there is no restriction.

https://github.com/paritytech/polkadot-sdk/blob/c8112e2c6fd9f1f5e7cb767b682fda746fc00520/substrate/frame/tips/src/lib.rs#L263-L265